src/Microsoft.DotNet.ProjectModel.Server/Program.cs

```diff
@@ -0,0 +1,164 @@
+        }
+
+        public static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            app.Name = "dotnet-projectmodel-server";
+            app.Description = ".NET Project Model Server";
+            app.FullName = ".NET Design Time Server";
+            app.Description = ".NET Design Time Server";
+            app.HelpOption("-?|-h|--help");
+
+            var verbose = app.Option("--verbose", "Verbose ouput", CommandOptionType.NoValue);
+            var hostpid = app.Option("--hostPid", "the process id of the host", CommandOptionType.SingleValue);
+            var hostname = app.Option("--hostName", "the process id of the host", CommandOptionType.SingleValue);
+            var port = app.Option("--port", "the communication port", CommandOptionType.SingleValue);
+
+            app.OnExecute(() =>
+            {
+                var loggerFactory = new LoggerFactory();
+                loggerFactory.AddConsole(verbose.HasValue() ? LogLevel.Debug : LogLevel.Information);
+
+                var logger = loggerFactory.CreateLogger<Program>();
+
+                if (!MonitorHostProcess(hostpid, logger))
+                {
+                    return 1;
+                }
+
+                var intPort = CheckPort(port, logger);
+                if (intPort == -1)
+                {
+                    return 1;
+                }
+
+                if (!hostname.HasValue())
+                {
+                    logger.LogError($"Option \"{hostname.LongName}\" is missing.");
+                    return 1;
+                }
+
+                var program = new Program(intPort, hostname.Value(), loggerFactory);
+                program.OpenChannel();
+
+                return 0;
+            });
+
+            return app.Execute(args);
+        }
+
+        public void OpenChannel()
+        {
+            var logger = _loggerFactory.CreateLogger($"OpenChannel");
+
+            // This fixes the mono incompatibility but ties it to ipv4 connections
+            _listenSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            _listenSocket.Bind(new IPEndPoint(IPAddress.Loopback, _port));
+            _listenSocket.Listen(10);
+
+            logger.LogInformation($"Process ID {Process.GetCurrentProcess().Id}");
+            logger.LogInformation($"Listening on port {_port}");
+
+            while (true)
+            {
+                var acceptSocket = _listenSocket.Accept();
+                logger.LogInformation($"Client accepted {acceptSocket.LocalEndPoint}");
+
+                var connection = new ConnectionContext(_hostName,
+                                                       acceptSocket,
+                                                       _protocolManager,
+                                                       _workspaceContext,
+                                                       _projectContextManagers,
```

Comment:
there is no random space here. `acceptSocket` doesn't start with `_`, just an illusion.